app/components/mapStats/common/MapStatsTypeSwitcher.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useContext } from 'react';
+import { Button, Card } from 'antd';
+import { MapInfo } from '../../../lib/api/apiRequests';
+import { AuthContext } from '../../../lib/contexts/AuthContext';
+import { cleanTMFormatting } from '../../../lib/utils/formatting';
+
+export enum MapStatsType {
+    GLOBAL = 'Global',
+    PERSONAL = 'Personal'
+}
+
+interface MapStatsTypeSwitcherProps {
+    mapStatsType: MapStatsType;
+    mapData: MapInfo;
+    toggleMapStatsType: () => void;
+}
+export const MapStatsTypeSwitcher = ({ mapStatsType, mapData, toggleMapStatsType } : MapStatsTypeSwitcherProps) => {
+    const { user, startAuthFlow } = useContext(AuthContext);
+
+    return (
+        <div className="flex flex-col gap-4 items-center">
+            <div className="text-xl">
+                You are currently viewing
+                {' '}
+                {mapStatsType === MapStatsType.GLOBAL ? 'the' : 'your'}
+                {' '}
+                <b>{mapStatsType}</b>
+                {' '}
+                statistics
+                {' '}
```

Comment:
Not a huge fan of this - could you maybe assemble the strings in a separate function and just put them in here?
The `<b>` will need to be on its own, but the rest looks a bit scuffed in the code.